File: 2125-number-of-laser-beams-in-a-bank/2125-number-of-laser-beams-in-a-bank.go
func numberOfBeams(banks []string) int {
    // prev represents the count of devices in previous row with non zero devices
    prevRowDeviceCount := 0
    ans := 0      
    for _, bank := range banks {
        if !strings.Contains(bank, "1") {
            continue
        }
        
        currentRowDeviceCount := countOnes(bank)
        ans += prevRowDeviceCount * currentRowDeviceCount
        prevRowDeviceCount = currentRowDeviceCount
    }
    return ans
}

func countOnes(bank string) int {
    currentRowDeviceCount := 0
    for _, ch := range bank {
        if(ch == '1') {
            currentRowDeviceCount++
        }
    }
    return currentRowDeviceCount
}

File: 2125-number-of-laser-beams-in-a-bank/2125-number-of-laser-beams-in-a-bank.cpp
class Solution {
public:
    int numberOfBeams(vector<string>& bank) {
        int m = bank.size();
        vector<int> count(m, 0);
        for(int i=0;i<m;i++) {
            for(auto ch:bank[i]) {
                if(ch == '1') {
                    count[i]++;
                }
            }
        }

        int prev = 0, ans = 0;        
        for(int i=0;i<m;i++) {
            // find the next non zero row
            if(count[i] != 0) {
                ans += prev * count[i];
                prev = count[i];
            }
        }
        return ans;
    }
};